lib/puppet/defaults.rb

```diff
@@ -854,11 +854,23 @@ def self.default_diffargs
         This is distinct from the certificate authority's CRL."
     },
     :certificate_revocation => {
-        :default  => true,
-        :type     => :boolean,
-        :desc     => "Whether certificate revocation should be supported by downloading a
-          Certificate Revocation List (CRL)
-          to all clients.  If enabled, CA chaining will almost definitely not work.",
+        :default  => 'chain',
+        :type     => :certificate_revocation,
+        :desc     => <<EOT
+Whether certificate revocation checking should be enabled, and what level of checking should be performed.
+
+When certificate_revocation is set to 'true' or 'chain', Puppet will download the CA CRL and will perform revocation
+checking against each certificate in the chain. When Puppet is using a single root certificate and has no intermediates
```

Comment:
Ah ok. So if I understand correctly this sentence:
> Puppet will download the CA CRL and will perform revocation checking against each certificate in the chain.

is not entirely currently true, because CRL checking against multiple CRLs doesn't currently work? 

If that's true, could we reword this to that effect? Something like:
```
When certificate revocation is set to 'true' or 'chain', Puppet will download the CA CRL and perform revocation checking against the CA cert. However, the CA cert cannot contain intermediate certificates when these settings are in use. If intermediate CAs are in use, revocation checking will fail due to current limitations in loading multiple CRLs. Thus the value of 'chain' is optimistic - it is a goal to support revocation checking of chained CAs with multiple associated CRLs in the future.
```